### PR TITLE
Library Object Updates [WIP]

### PR DESCRIPTION
**Pull request by espg**
Still working on this, but, basically does the following:

 1. Changes object construction and naming conventions
 2. Moves and standardizes data file locations
 3. Changes user interaction with the indices

The biggest change from a UX perspective is how indicies are called:

```
from geostack import SpatialIndexLS8
ls8 = SpatialIndexLS8()
matches = ls8.query_pathrow(69, -50)
```

When `SpatialIndexLS8` is instantiated, it pulls in the intake file for download paths, the balltree for queries, and the csv file with footprint info while making the object. When `query_pathrow` is called, it subsets the pandas dataframe using the matches, and returns a duplicate object that is a different subclass of `SpatialIndex` called `Search`. My thought is that this class will have a `__repr__` method to display the query point and footprints using ipyleaflet (probably with a helper function to convert the footprints to a geojson multipolygon). The object will also have a fetch method that grabs either all the data, or subset... which will have to have a way to note dates.

Lots of work still to be done, but I'll start on getting the intake fetch to work. @whyjz any thoughts on parsing the dates? I'm thinking of maybe a function to estimate valid dates, but we could also take the empirical dates from the larger csv file from landsat...

### CODE REVIEW

File: geostacks/geostacks.py [hunk 1]
```diff
@@ -11,78 +11,43 @@
 from datetime import datetime
 
 
-class SpatialIndex:
+class SpatialIndexL3:
```
Comment (whyjz) [inline geostacks/geostacks.py:14]: What does `L3` mean?

**Comment by whyjz** [comment]
Cool Thanks @espg! There is an unused method called `search_s3` in `geostacks.py` which queries available dates from the S3 bucket. This is from the EZTrack project so I think we have to modify that a bit to make it work. Maybe we can build a clove cover filter based on that...

Looks like `__init__.py` has to be modified for geostacks to be correctly imported. Let me see if I can directly push to this branch.

**Comment by whyjz** [comment]
Looks like there are still some other bugs... maybe this is great timing to set up a Github Action to automatically check on this.
ScrSelection_019

**Comment by whyjz** [comment]
And also I wonder if we need a more specific name for the footprint info than `.data`...

File: geostacks/geostacks.py (hunk 1 above)
Comment (espg) [reply geostacks/geostacks.py:14]: Level 3 data -- i.e., gridded data scenes. The thought is that the indexing scheme will be similar (static footprint definitions in a data structure) for things like gridded MODIS and NCEP data, but different enough for a new class definition for swath data like non-gridded MODIS, or similar

**Comment by espg** [comment]
I think the error above is just a bit different behavior... instead of 

```
print(idxs)
ls8.data.loc[idxs]
```

You just want

```
idxs 
ls8.data.loc[idxs.data.index]
```

(I think)

@whyjz

File: geostacks/geostacks.py (hunk 1 above)
Comment (whyjz) [reply geostacks/geostacks.py:14]: I thought we are working with Landsat Level 1 data but I get what you mean!

**Pull request merged**